warehouse/query-core/src/main/java/datawave/query/jexl/functions/QueryFunctionsDescriptor.java

```diff
@@ -61,17 +64,17 @@ public JexlNode getIndexQuery(ShardQueryConfiguration config, MetadataHelper hel
                                     + args.get(2).image + '}');
                 case QueryFunctions.MATCH_REGEX:
                     // Return an index query.
-                    return getIndexQuery();
+                    return getIndexQuery(helper, datatypeFilter);
                 case INCLUDE_TEXT:
                     // Return the appropriate index query.
-                    return getTextIndexQuery();
+                    return getTextIndexQuery(helper, datatypeFilter);
```

Comment:
Maybe instead of passing down the metadata helper and the data type filter, you could run the field filtering prior to these calls and just pass the resulting fields down.

Reply:
Changed to calculating the fields before sending to method